src/main/java/ro/utcn/daneeee/assignment1/persistance/memory/InMemoryQuestionRepository.java

```diff
@@ -0,0 +1,74 @@
+package ro.utcn.daneeee.assignment1.persistance.memory;
+
+import org.springframework.stereotype.Repository;
+import ro.utcn.daneeee.assignment1.model.Question;
+import ro.utcn.daneeee.assignment1.persistance.QuestionRepository;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class InMemoryQuestionRepository implements QuestionRepository {
+
+    private final Map<Integer, Question> data = new ConcurrentHashMap<>();
+    private final AtomicInteger currentId = new AtomicInteger(0);
+
+    @Override
+    public Question save(Question question) {
+        if(question.getId() == null){
+            question.setId(currentId.incrementAndGet());
+        }
+        data.put(question.getId(), question);
+        return question;
+    }
+
+    @Override
+    public Optional<Question> findById(int id) {
+        return Optional.ofNullable(data.get(id));
+    }
+
+    @Override
+    public Optional<Question> findByTitle(String text){
+        List<Question> questions = findAll();
+        for(Question currentQuestion: questions){
+            if(currentQuestion.getTitle().equals(text)){
+                return Optional.ofNullable(currentQuestion);
+            }
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public void remove(Question question) {
+        data.remove(question.getId());
+    }
+
+    @Override
+    public List<Question> findAll() {
+        List<Question> list = new ArrayList<>(data.values());
+        list.sort(new Comparator<Question>() {
```

Comment:
You could replace this big comparator anonymous function by building it like so: `Comparator.comparing(Question::getCreationDate).thenComparing(Question::getId)`.